### PR TITLE
Update jax, brax and flax versions (fixes the jax.tree_util warnings)

### PR DESCRIPTION
**Pull request by Lookatator**
Related to issue #74 

- all jax tree-based functions are now imported from the tree_util module
- updates the dependencies in requirements.txt and setup.py
- Update the version of Brax in the requirements and in the setup.py

--------------

This PR also adds the wrapper `CompletedEvalWrapper` that used to be in Brax, but has been removed in the most recent versions.

In summary, this wrapper used to be present in Brax, and we rely on some elements of it in 4 algorithms (I think those are: SAC, DIAYN, DADS and TD3), and their tests where all failing for that reason.

So what I did is copying the old wrapper of Brax here, and now all the tests pass.

This is a provisional solution to make it work in the same way as before.
Maybe we can improve and adapt our own code to the new structure of QDax, but I think that is for another time/PR.

### CODE REVIEW

**Comment by limbryan** [comment]
We should also update flax and chex version to fix the warnings completely - Flax (0.6.0) and chex (0.1.4) versions have these changes updated.

**Comment by Lookatator** [comment]
> We should also update flax and chex version to fix the warnings completely - Flax (0.6.0) and chex (0.1.4) versions have these changes updated.

That was included in the PR. 

However, I've just changed it a bit. Because:
- flax == 0.6.0 requires jax >= 0.3.16
- jax >= 0.3.16 does not have tree_multimap
- all versions of brax do not have tree_multimap

https://github.com/adaptive-intelligent-robotics/QDax/blob/8aa72354ba397b2005657efc94c20f9d27f10247/setup.py#L27-L29

**Comment by felixchalumeau** [comment]
Hey :wave: 

Are the requirements update necessary to remove the warnings?

If yes, have you made a few double check to be sure that those updates won't affect performances? I am always afraid with version updating, it can often bring bad surprises.

File: qdax/environments/wrappers.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+from typing import Dict
```
Comment (felixchalumeau) [inline qdax/environments/wrappers.py:1]: Is this wrapper really linked to that particular PR? Is there no way to make a separate PR for this wrapper?

**Comment by felixchalumeau** [comment]
Hi guys,

I reviewed the PR and am happy with it except for the `CompletedEvalWrapper`.
I want to ask you more questions about it and also, does it has to go with this PR in term of compatibility or can it be introduced in a separate pull request?

File: qdax/environments/wrappers.py (hunk 1 above)
Comment (Lookatator) [reply qdax/environments/wrappers.py:1]: Yes, this wrapper is really needed.

In summary, this wrapper used to be present in Brax, and we rely on some elements of it in 4 algorithms (I think those are: SAC, DIAYN, DADS and TD3), and their tests where all failing for that reason.

So what I did is copying the old wrapper of Brax here, and now all the tests pass.

This is a provisional solution to make it work in the same way as before.
Maybe we can improve and adapt our own code to the new structure of QDax, but I think that is for another time/PR.

Comment (felixchalumeau) [reply qdax/environments/wrappers.py:1]: Ok! Thanks for your answer! Could you add this in the PR description?

**Review by limbryan — APPROVED** [review]
